### PR TITLE
DX: replace nbstripout by nb-clean

### LINKED ISSUE #88: Switch from nbstripout to nbclean
See also https://github.com/ComPWA/policy/issues/352

_Originally posted by @redeboer in https://github.com/ComPWA/gluex-nstar/pull/87#discussion_r1723611935_

### PR DESCRIPTION
**Pull request by shenvitor**
https://github.com/srstevenson/nb-clean

- Closes #88

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -33,17 +33,16 @@ repos:
...
-          - --extra-keys
-          - |
-            cell.attachments
-            cell.metadata.editable
-            cell.metadata.id
-            cell.metadata.slideshow
+          - --remove-empty-cells
+          - --preserve-cell-metadata
+          - tags
+          - slideshow
```
Comment (redeboer) [inline .pre-commit-config.yaml:44]: I would also remove `slideshow`. Only keep `tags` (which are used by `myst-nb`)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -33,17 +33,16 @@ repos:
...
         args:
-          - --extra-keys
-          - |
-            cell.attachments
-            cell.metadata.editable
-            cell.metadata.id
-            cell.metadata.slideshow
+          - --remove-empty-cells
+          - --preserve-cell-metadata
+          - tags
```
Comment (redeboer) [inline .pre-commit-config.yaml:43]: Also keep `jupyter` and `mystnb` (see 4420106723e3560b8f1754e7ee442ac684ddff9a)

**Review by redeboer — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (shenvitor) [reply .pre-commit-config.yaml:44]: ok

**Comment by shenvitor** [comment]
Change strategy:
https://github.com/kynan/nbstripout?tab=readme-ov-file#removing-empty-cells

seems like it is ok to remove empty cells in nbstripout too, so I will try

**Comment by shenvitor** [comment]
Change strategy and replaced by new PR #92

**Pull request closed without merging**